src/utils/geoserverUtils.tsx

```diff
@@ -0,0 +1,123 @@
+import ImageWMS from 'ol/source/ImageWMS';
+import Source from 'ol/source/Source';
+import TileWMS from 'ol/source/TileWMS';
+
+import { Logger } from '@terrestris/base-util';
+import { getBearerTokenHeader } from '@terrestris/shogun-util/dist/security/getBearerTokenHeader';
+import { SHOGunAPIClient } from '@terrestris/shogun-util/dist/service/SHOGunAPIClient';
+
+export type WorkspaceInfo = {
+  workspace: string;
+  layerName: string;
+};
+
+export const fetchWorkspaceFromGetCapabilities = async (
+  layerUrl: string,
+  client: SHOGunAPIClient | null
+): Promise<WorkspaceInfo | undefined> => {
+  try {
+    const url = `${layerUrl}SERVICE=WMS&REQUEST=GetCapabilities`;
+    const response = await fetch(url, {
+      headers: {
+        ...getBearerTokenHeader(client?.getKeycloak())
+      }
+    });
+
+    const xml = await response.text();
+    const parser = new DOMParser();
+    const xmlDocument = parser.parseFromString(xml, 'text/xml');
+    const layers = xmlDocument.getElementsByTagName('Layer');
+
+    let output: WorkspaceInfo | undefined;
+    Array.from(layers).forEach((layerElement) => {
+      const name = layerElement.getElementsByTagName('Name')[0]?.textContent;
+
+      if (name) {
+        const [workspace, layerName] = name.split(':');
+        output = {
+          workspace,
+          layerName
+        };
+      }
+    });
+
+    return output;
+  } catch (error) {
+    Logger.error('Error: ', error);
+    return undefined;
+  }
+};
+
+const fetchSld = async (
+  name: string,
+  layerUrl: string,
+  client: SHOGunAPIClient | null
+): Promise<string | undefined> => {
+  try {
+    const cleanLayerUrl = layerUrl.replace('/ows?', '');
+    const url = `${cleanLayerUrl}/rest/styles/${name}.sld`;
+
+    const response = await fetch(url, {
+      headers: {
+        ...getBearerTokenHeader(client?.getKeycloak())
+      }
+    });
+
+    const sldData = await response.text();
+    if (!sldData) {
+      return undefined;
+    }
+    return sldData;
+  } catch (error) {
+    Logger.error('Error: ', error);
+    return undefined;
+  }
+};
+
+export const fetchGeoserverStyle = async (
+  layerWorkspace: string,
+  layerName: string,
+  layerUrl: string,
+  client: SHOGunAPIClient | null
+): Promise<string | undefined> => {
+  const cleanLayerUrl = layerUrl.replace('/ows?', '');
+  const url = `${cleanLayerUrl}/rest/layers/${layerWorkspace}:${layerName}.json`;
```

Comment:
This URL (and also the one used in `fetchSld()`) is usually accessible for users holding the role `ROLE_ADMINISTRATOR` in GeoServer only.